Source/DafnyCore/Dafny.atg

```diff
@@ -144,6 +144,7 @@ TOKENS
   by = "by".
   in = "in".
   decreases = "decreases".
+  nonincreases = "nonincreases".
```

Comment:
Now that it's there, we might as well keep it, but was there a strong reason to add it?

Reply:
Sometimes the check that Dafny needs to perform is, roughly speaking, <= instead of <. To be able to express those conditions, it seemed most natural to have a form of the expression that encoded <= instead of <. It would be possible, alternatively, to have `(x decreases to y) || x == y`, but it seems a little more awkward, and I think would be more difficult to implement, as well (though I'm no longer sure of that).